validation - index only fields found within filter function

skyler-kv opened the pull request:
fixes #2873 

changes the name of `IncludeExcludeIndexFieldsRule` and changes check for index fields to be index only fields

skyler-kv committed:
changes error message

billoley committed:
Specify ResultSet.TYPE_SCROLL_INSENSITIVE in CachedRunningQuery when we need to use certain ResultSet methods (#2953)

skyler-kv committed:
Revert "changes error message"

This reverts commit 991bbf50a9ef98fc9d0efe70d8ef71e69d606ab9.

skyler-kv committed:
changes error message

FineAndDandy committed:
RemoveGroupingContext should return all attributes of the document (#2884) (#2900)

rdhayes68 committed:
Update tests to provide a more specific seek range so the initial seek avoids loading unnecessary data. (#2947)

jalphonso committed:
create new data manager per test class per JUnit runner (#2960)

Co-authored-by: Joe Alphonso <jfalpho@uwe.nsa.gov>

avgAGB committed:
Adding utf8 charset conversions (#2766)

* Adding utf8 charset conversions

* Tweaking imports

* Getting commit consistent with checkstyle

* Adding back in date range filtered

* Disabled for Junit5 has replaced ignore

* Setting up constant for TextUtil

* Updating references in microservices

* Fixing spotbugs

* Another fix for RefValue

* GitHub Actions: Fix Formatting

---------

Co-authored-by: Adam Lerman <awlerma@uwe.nsa.gov>
Co-authored-by: GitHub Actions <datawave@github.com>

dws4 committed:
task/descriptionPopulate: Fixed Column for Description issue (#2955)

* Fixed Column for Description issue

* Updating git-ignore to include .maven-dockerignore

* Adding extra comments for the parseVal function

foster33 committed:
Change formatter to fail on diff instead of auto commit (#2965)

* Change formatter to fail on diff detected

* Simulated format fix

apmoriarty committed:
ScanResult now supports partial intersections. (#2956)

* ScanResult now supports partial intersections.

A partial intersection is possible when the left hand set is complete
and the right hand side is guaranteed to be in sorted order (i.e., an
Equality term).

Feature flag partial intersection logic

* Enforce logic order with parens

* ScanResult should correctly intepret flag to allow partial intersections

apmoriarty committed:
Correct scan hints for candidate search and retrieval (#2967)

FineAndDandy committed:
Feature/splittable rfiles (#2208)

* WIP: splittable rfile input format

* Added unit tests

* splittable rfile input format fixes and updates

* code review fixes

* sonar lint fix

* Prevent pre-splitting of files by FileInputFormat

* Move RFileUtil to datawave-core and use in SplittableRFileRecordReader

skyler-kv committed:
changes indexed fields rule to be index only and renamed IncludeExcludeIndexFieldsRule

skyler-kv committed:
formatting

foster33 commented on warehouse/query-core/src/test/java/datawave/query/tables/ShardQueryLogicQueryValidationTest.java:
```diff
@@ -172,7 +172,7 @@ public void testBeanCreation() {
         expectedRules.add(new FieldPatternPresenceRule("Check Presence of Field or Pattern", Map.of("_ANYFIELD_", "Detected presence of _ANYFIELD_"),
                         Map.of(".*", "Detected pattern '.*' that will match everything")));
-        expectedRules.add(new IncludeExcludeIndexFieldsRule("Check #INCLUDE and #EXCLUDE for Indexed Fields"));
+        expectedRules.add(new IncludeExcludeIndexOnlyFieldsRule("Check #INCLUDE and #EXCLUDE for Indexed Fields"));
```
```suggestion
        expectedRules.add(new IncludeExcludeIndexOnlyFieldsRule("Check #INCLUDE and #EXCLUDE for Index Only Fields"));
```

skyler-kv committed:
fixes description for name change

Co-authored-by: foster33 <84727868+foster33@users.noreply.github.com>

kmtoth committed:
Update pom's for 7.25.0-SNAPSHOT

emiliodskinner committed:
task-2628-core: Update Logging to SLF4j in datawave-warehouse, A to I (#2693)

* task-2628-core: Update Logging to SLF4j in datawave-warehouse, A to I

* GitHub Actions: Fix Formatting

* task-2628-core: fix formatting

* task-2628-core: IngestJob.java

* GitHub Actions: Fix Formatting

* Remove unused imports (#2628)

---------

Co-authored-by: GitHub Actions <datawave@github.com>

skyler-kv committed:
adds new value to IncludeExcludeIndexOnlyFieldsRule

drewfarris committed:
More bugfixes for tag clouds / keyword extraction (#2970)

* Fixed issue where the max tags per cloud parameter was not propogated to the backend.
* Fixed issue where a small page size on the id lookup portion of the query led to the creation of multiple tag clouds for a single language.
* Fixed issue where the language used for keyword extraction was used to group tag clouds instead of the language marked on the input document
* Fixed issue where we couldn't input the required number of document ids for tag clouds because lookup was limited to 100 ids. Tag cloud lookup can now perform to 500 lookups instead.

emiliodskinner committed:
task-2628: Update Logging to SLF4j in datawave-warehouse/ingest-core pt1 (#2696)

* task-2628-ingest-core: warehouse/ingest-core pt1

* task-2628-ingest-core: warehouse/ingest-core pt1

* Apply suggestions from code review

Co-authored-by: foster33 <84727868+foster33@users.noreply.github.com>

* Resolve merge conflicts (#2628)

* Remove log conditionals (#2628)

---------

Co-authored-by: Moriarty <22225248+apmoriarty@users.noreply.github.com>
Co-authored-by: foster33 <84727868+foster33@users.noreply.github.com>

emiliodskinner committed:
Revert "Update Logging to SLF4j in datawave-warehouse, A to I" (#2628) (#2974)

alerman committed:
Add release through CI

alerman committed:
Add deploy and build and push images step

alerman committed:
Make sure right dir and right tag checked out

jalphonso committed:
temporarily disable quickstart-build-and-test ci job (#2980)

Co-authored-by: Joe Alphonso <jfalpho@uwe.nsa.gov>

apmoriarty committed:
Upgrade query metric service version of datawave from 7.18.3 to 7.24.0 (#2975)

apmoriarty committed:
DocumentIterator can now aggregate non-event fields (#2968)

Support for index only and term frequency fragments, limited to equality
terms (i.e., regex/range not supported yet).

Clean up DocumentIteratorOption parsing

alerman committed:
Update per pr comments

alerman committed:
Updating auto release pipeline

alerman committed:
Updating auto release pipeline

alerman committed:
Skip Tests and Checks for release pipeline

jalphonso committed:
update quickstart maven binary (#2983)

Co-authored-by: Joe Alphonso <jfalpho@uwe.nsa.gov>

alerman committed:
Changes needed to cleanup releases (#2984)

* Changes needed to cleanup releases

* Update microservices/microservice-parent/pom.xml

* Fix Formatting

drewfarris committed:
More stream cleanup in YakeKeywordExtractor (#2971)

* The stream api usage in calculateCandidateKeywords was hard to read, refactored to use loops for clarity.

apmoriarty committed:
DocumentScheduler now supports composite fields (#2982)

Mark record id attribute as 'to keep'

Use ValueToAttributes to generates composite fields during event
aggregation. ValueToAttributes is now reusable.

DocumentIterator now supports retaining grouping notation

Refactor DocumentIterator so that key parsers and aggregation related
classes are now in the base DocumentIteratorOptions class.

lbschanno committed:
Update error message from AmbiguousNotRule (#2959)

Update the error message returned by the AmbiguousNotRule to avoid users
assuming that the suggested syntax is overall correct and appropriate
for copy/pasting.

Closes #2939

apmoriarty committed:
TypeAttribute hashcode performance improved (#2976)

The result of a hashcode calculation is now cached, along with the
delegate string

The hashcode is now serialized to avoid expensive calls when
deserializing TypeAttributes

apmoriarty committed:
Optimize Numeric Serialization (#2977)

* Numeric now writes normalized and non-normalized values, reduces read times by almost half

* Document utilizes class cache when deserializing

alerman committed:
Add ability to release microservices through ci (#2985)

alerman committed:
Update microservice-release.yml

alerman committed:
Fix docker prefix

foster33 committed:
Update zookeeper version used in datawave quickstart (#2972)

* Update zookeeper version in datawave quickstart

* Update version in bogus zk URL

apmoriarty committed:
Attribute provides means to cache hashcode and sizeInBytes computation (#2986)

Most implementations of Attribute now cache the computation for hashcode
and computation.

Fix bug in JexlASTHelper's deconstructIdentifier method that caused a
new String to be built, even when not necessary.

alerman committed:
Add scm id for microservices (#2989)

* Do not create new branch

* [maven-release-plugin] prepare release microservices/microservice-parent-4.0.4

* [maven-release-plugin] prepare for next development iteration

* inclide settings file in args

* Update all dependent ms to use the new parent

* [maven-release-plugin] prepare release microservices/microservice-service-parent-5.0.4-RC1

* [maven-release-plugin] prepare for next development iteration

* [maven-release-plugin] prepare release microservices/microservice-service-parent-5.0.4

* [maven-release-plugin] prepare for next development iteration

* Update all services to use new parent

* Use tokens with write access

* Revert "Use tokens with write access"

This reverts commit eea89165aab77e48562b99c51cc7a0326dabbad8.

* Use tokens with write access

* [maven-release-plugin] prepare release microservices/microservice-parent-4.0.5-CITEST-4

* [maven-release-plugin] prepare for next development iteration

---------

Co-authored-by: alerman <datawave@github.com>

apmoriarty committed:
DocumentIterator no longer aggregates all candidates prior to returning the first result (#2987)

Commit:
[maven-release-plugin] prepare release microservices/services/query-metric/service-4.1.3

Commit:
[maven-release-plugin] prepare for next development iteration

billoley committed:
Query metric api version of datawave dependency to 7.24.0

Commit:
[maven-release-plugin] prepare release microservices/services/dictionary/service-4.0.7

Commit:
[maven-release-plugin] prepare for next development iteration

apmoriarty committed:
Serialize the index instead of the full Attribute class name (#2990)

hgklohr committed:
Update pom's for 7.26.0-SNAPSHOT

alerman committed:
Add readme for releasing (#2995)

rdhayes68 committed:
Update iterators to provide a more specific seek range so the initial seek avoids loding unnecessary data. (#2994)

Co-authored-by: Reggie Hayes rdhaye3 <rdhaye3@ip-10-113-13-151.evoforge.org>

kmtoth committed:
Upgrade query metric service and api version of datawave from 7.24.0 to 7.25.0 (#2997)

Commit:
[maven-release-plugin] prepare release microservices/services/query-metric/api-4.1.1

Commit:
[maven-release-plugin] prepare for next development iteration

drewfarris committed:
Implements additional fixes for tag cloud generation (#2991)

* Improves number detection in YakeKeywordExtractor

* Avoids exceptions in Float.parseFloat and an expensive string replace to remove number separators.
* Improves number separator detection for multiple locales (e.g., those that don't use commas as place separators in numbers)

* Fixed LookupUUIDUtil.generateTagCloud to close query executor when done

* Cache stopwords iterator-side.

* Avoid multiple reloads of a classpath resource for stopwords by caching the lists on a per-language basis.

* Fixes issue where tag document frequency didn't match the unique source count

kmtoth committed:
Upgrade query metric api version to 4.1.1 (#3000)

alerman committed:
Break up microservice testing and main repo testin (#2992)

* Break up microservice testing and main repo testin

* Add files that hadnt been added

* Update ms test name

* Update ms test name

* Update workflow file names to fix concurrency

* Add negations for files to ignore

* Add no-op tests so we can set branch protections

Commit:
[maven-release-plugin] prepare release microservices/services/query-metric/service-4.1.4

Commit:
[maven-release-plugin] prepare for next development iteration

apmoriarty committed:
Improve CountMap deserialization speed (#3002)

alerman committed:
Allow keyFunctor until accumulo makes a public api update (#2473)

Co-authored-by: Daniel Roberts <ddanielr@gmail.com>

avgAGB committed:
Refactoring StringUtils calls to java builtin (#2733)

* Drafting builtin function refactor

* Additional updates

* More refactoring

* Adding more builtins

* Working through some query core references

* More split updates

* Formatting issue

* More additions

* split refactor

* Checking particular regex case test

* Adding in unique fields update

* Update for term frequency query transform

* ingest/query core tweaks

* Another update before tweaking some regexes

* Ingest core updates

* Working through some more split calls

* Ingest core work

* More work in query core

* More work on query core

* tweaking formatting

* Adding in more space swaps

* Cleaning up empty strings

* More adjustments for spacing

* More tweaks for spaces where applicable

* Additional formatting

* More formatting

* Tweaking pipe refs

* Updating calls for null chars

* ExcerptFields commit

* Going to extract the spacing changes into a different set of commits

* Removing unused imports

* Fixing unused imports

* Removing unused imports

* Missing an import removal

* Rebase, undoing some of the regex edits

* Fixing formatting, ignoring split brain

* Removing the char tweaks for another MR

* Pulling back spacing

* Reverting more space changes

* More edits for the spacing

* Remove spacing change in RunningQuery

* Some updates to utils calls

* Updating split calls

* Updating unit test refs

* GitHub Actions: Fix Formatting

* Removing unused call in StatsHyperLogMapper

* Merging in integration

* Returning param sep to comma for pipeline build

---------

Co-authored-by: GitHub Actions <datawave@github.com>

apmoriarty committed:
KryoDocumentSerializer and KryoDocumentDeserializer now use direct methods (#3008)

lbschanno committed:
Add date capabilities to #GROUPBY (#2731)

* Add date capabilities to #GROUPBY

Add the ability to subselect part of a date in the #GROUPBY function.
These can now be specified similar to the #UNIQUE function, and support
the granularities MILLISECOND, SECOND, TENTH_OF_HOUR, HOUR, DAY, MONTH,
and YEAR. In addition, the following functions have been added:

GROUPBY_MILLISECOND()
GROUPBY_SECOND()
GROUPBY_TENTH_OF_HOUR()
GROUPBY_HOUR()
GROUPBY_DAY()
GROUPBY_MONTH()
GROUPBY_YEAR()

Closes #2703

* Add documentation and parsing tests

* Remove unused imports

* Ensure truncated values are returned

* Ensure GroupBy.validate() does not swallow exception

* GitHub Actions: Fix Formatting

* Update warehouse/query-core/src/test/java/datawave/query/language/functions/jexl/GroupByTest.java

Co-authored-by: foster33 <84727868+foster33@users.noreply.github.com>

---------

Co-authored-by: GitHub Actions <datawave@github.com>
Co-authored-by: foster33 <84727868+foster33@users.noreply.github.com>

apmoriarty committed:
Document now caches the row to timestamp operation (#3006)

avgAGB committed:
Adding in conditional for formatting updates (#3011)

* Adding in conditional for formatting

* Updating other unit tests

apmoriarty committed:
Improve speed of document creation by avoiding expensive text decode (#3007)

lbschanno committed:
Refactor AccumuloSyntaxParserConstants so it is not an interface (#2740)

Refactor AccumuloSyntaxParserConstants so that it is not an interface
that consists solely of constants, but is rather a final, uninstantiable
class with constants. While not strictly an issue, it is odd for
AccumuloSyntaxParser and AccumuloSyntaxParserTokenManager to both be
considered implementations of AccumuloSyntaxParserConstants.

Closes #2739

Co-authored-by: foster33 <84727868+foster33@users.noreply.github.com>

apmoriarty committed:
Add grouping tests from branch to QueryIteratorIT (#3012)

Fix bug in matchesInGroup that prevented matches when an index higher
than zero was requested.

ivakegg committed:
Revert "Refactoring StringUtils calls to java builtin (#2733)" (#3017)

This reverts commit 8133a6c993f6caf7d21eaa8b03bec36d18a11ba3.

apmoriarty committed:
ExpandMultiNormalizedTerms will drop invalid ranges and throw an exception if no valid range exists after applying normalizers (#3014)

keith-turner committed:
Adds a new shard table balancer (#2978)

* Adds a new shard table balancer

Adds a new optional balancer for the shard table that supports balancing
by host then tserver using rendevous hashing.  See the class level javadoc
on ShardRendezvousHostBalancer for an overview of what the balancer
offers.

Co-Authored-By: Daniel Roberts ddanielr <ddanielr@gmail.com>

* Add javadoc comments

* Update warehouse/accumulo-extensions/src/main/java/datawave/ingest/table/balancer/RendezvousHostBalancer.java

Co-authored-by: Daniel Roberts <ddanielr@gmail.com>

* Update warehouse/accumulo-extensions/src/main/java/datawave/ingest/table/balancer/RendezvousHostBalancer.java

Co-authored-by: Daniel Roberts <ddanielr@gmail.com>

* Update warehouse/accumulo-extensions/src/main/java/datawave/ingest/table/balancer/RendezvousHostBalancer.java

Co-authored-by: Daniel Roberts <ddanielr@gmail.com>

* Update warehouse/accumulo-extensions/src/main/java/datawave/ingest/table/balancer/RendezvousHostBalancer.java

Co-authored-by: Daniel Roberts <ddanielr@gmail.com>

* Update warehouse/accumulo-extensions/src/main/java/datawave/ingest/table/balancer/ShardRendezvousHostBalancer.java

Co-authored-by: Daniel Roberts <ddanielr@gmail.com>

* update javadoc

* use all of the tservers

* improve comment

* Update warehouse/accumulo-extensions/src/main/java/datawave/ingest/table/balancer/RendezvousHostBalancer.java

Co-authored-by: P.A. <lionrugi@gmail.com>

* Update warehouse/accumulo-extensions/src/main/java/datawave/ingest/table/balancer/RendezvousHostBalancer.java

Co-authored-by: P.A. <lionrugi@gmail.com>

* Update warehouse/accumulo-extensions/src/main/java/datawave/ingest/table/balancer/RendezvousHostBalancer.java

Co-authored-by: P.A. <lionrugi@gmail.com>

* Update warehouse/accumulo-extensions/src/main/java/datawave/ingest/table/balancer/RendezvousHostBalancer.java

Co-authored-by: P.A. <lionrugi@gmail.com>

* test partition of size zero

---------

Co-authored-by: Daniel Roberts ddanielr <ddanielr@gmail.com>
Co-authored-by: P.A. <lionrugi@gmail.com>

ivakegg committed:
fixes #3015: updated to avoid tokenizing regex expressions (#3018)

* fixes #3015: updated to avoid tokenizing regex expressions

* Added test case to verify fix

apmoriarty committed:
Revert/bounded range normalization (#3020)

* Revert "ExpandMultiNormalizedTerms will drop invalid ranges and throw an exception if no valid range exists after applying normalizers (#3014)"

This reverts commit f2d2d2cb55d7fd15a5704cea7b8f93cb46e153ad.

* Remove bounded range check from IndexExpansion

hgklohr committed:
Update pom's for 7.27.0-SNAPSHOT

apmoriarty committed:
Feature flag option pruning (#2054)

apmoriarty committed:
Support grouping functions in TLD (#3019)

* Add support for grouping functions in TLD, prevent matching across children

* Add null checks

lbschanno committed:
Update deprecated usages of SequenceFile.createWriter() (#2499)

* Update deprecated usages of SequenceFile.createWriter()

Update several deprecated calls to SequenceFile.createWriter() to use
SequenceFile.createWriter(Configuration conf,
SequenceFile.Writer.Option... opts) instead.

Part of #2443

* Do not make paths qualified before creating writer

---------

Co-authored-by: Ivan Bella <ivan@bella.name>
Co-authored-by: foster33 <84727868+foster33@users.noreply.github.com>

dws4 committed:
task/expandableRows: Create neater drop down variations for Data Type on Dictionary (#2999)

* Making typed columns

* Removing console logging

* Fixing using row duplicates

* Putting regex as global var

* Update microservices/services/dictionary/service/src/main/frontend/src/pages/DataDictionary.vue

Removing whitespace

Co-authored-by: foster33 <84727868+foster33@users.noreply.github.com>

* Fixing description name

---------

Co-authored-by: foster33 <84727868+foster33@users.noreply.github.com>

ivakegg committed:
fixes #2935: Invoke IngestTypePruningVisitor in the VisitorFunction (#3016)

Co-authored-by: Moriarty <22225248+apmoriarty@users.noreply.github.com>

billoley committed:
Skip advance logic in AncestorQueryIterator.seek and TLDQueryIterator.seek when range startKey contains YIELD_AT_BEGIN marker (#3024)

apmoriarty committed:
Add integration test that exercises random document retrieval (#3023)

* Add SizesTest that exercises retrieval of many random over a small number of shards

* Migrate event from map to multimap

apmoriarty committed:
Add test for multiple normalizers (#3029)

* Add MultiNormalizerTest to simulate when multiple normalizers are
applied to a field across date ranges.

Refactor common methods to AbstractQueryTest

* Expect problems with text ranges

* Formatting

kmtoth committed:
Upgrade query metric service version of datawave to 7.26.1 (#3034)

lbschanno committed:
Validate Unique and Grouping temporal fields (#3022)

* Validate Unique and Grouping temporal fields

Ensure that when fields used with the #UNIQUE or #GROUP_BY function, if
they are paired with a temporal granularity other than ALL, that they
are datetime fields.

Fixes #3010

* Fix code formatting

* Fix javadoc formatting

* Remove unecessary braces

Commit:
[maven-release-plugin] prepare release microservices/services/query-metric/service-4.1.5

Commit:
[maven-release-plugin] prepare for next development iteration

apmoriarty committed:
Optionally limit the number of lines printed when streaming queries from the global index (#2238)

apmoriarty committed:
Add NodeTransformRule to pushdown regex terms known to fail index expansion (#2657)

SethSmucker committed:
Replace deprecated OptionBuilder with Option.builder() (#2488)

* Update in AccumuloCliOptions

* Update in GenerateMultipleNumShardCacheFile

* Update in GenerateSplitsFile

* Update in MetricsOptions

* Formatting

* Refactored SUFFIEX -> SUFFIX in GenerateMultipleNumShardsCacheFile

* Remove required from CONFIG_SUFFIX_OVERRIDE

* fix imports

---------

Co-authored-by: Ivan Bella <347158+ivakegg@users.noreply.github.com>

jack-gitdev committed:
Ssdeep overlap scorer update (#3009)

* Add cache for ngrams of query hashes. Prepare SSDeepNgramOverlapScorer to use cache.

* Revert "Add cache for ngrams of query hashes. Prepare SSDeepNgramOverlapScorer to use cache."

This reverts commit 7c9f386aec137e67f6d366ef2d2b55ef4d447858.

* Revert cache changes as they are not useful for overlap scoring. Overlap scoring changes in progress.

* Calculate overlapping ngrams via substring comparison

* Apply formatting to ssdeep overlap scoring changes

* Remove unused import in SSDeepNGramOverlapScorer

* Change location of ngram overlap calculation method and add recommended efficieny changes.

lbschanno committed:
Replace all usages of Accumulo Pair class (#3032)

Replace all usages of the class org.apache.accumulo.core.util.Pair class
with org.apache.commons.lang3.tuple.Pair. Additionally, add an
IllegalImport check to the maven checkstyle plugin configuration that
will explicitly forbid the use of the accumulo Pair class.

Fixes #3031

drewfarris committed:
Allows JSON POSTs to Datawave Query API (#3033)

Implements a Jackson deserializer that can convert a simple flat JSON object
containing one or more key/value pairs into the JAX-RS MultivaluedMap object
consumed in the Datawave QueryExecutorBean.

Includes a quickstart web test for validating POST with JSON.

apmoriarty committed:
Producer/Consumer Buffer (#3036)

* WIP: ThreadedRangeBundlerIterator without polling

* Extract main method to unit test, small cleanup

---------

Co-authored-by: FineAndDandy <cwilliams.work@gmail.com>

jalphonso committed:
bump maven version for quickstart and add aether settings (#3043)

* bump maven version for quickstart cdn downloads

* remove maven tarball sanity check

* add aether settings to github workflows

---------

Co-authored-by: Joe Alphonso <jfalpho@uwe.nsa.gov>

dws4 committed:
Adding URL Search Query Implementation (#3028)

* Adding URL Search Query Implementation

* Update microservices/services/dictionary/service/src/main/frontend/src/pages/DataDictionary.vue

Co-authored-by: jsduncan2 <jsduncan2@users.noreply.github.com>

* Update microservices/services/dictionary/service/src/main/frontend/src/pages/DataDictionary.vue

Co-authored-by: jsduncan2 <jsduncan2@users.noreply.github.com>

* Fixing search on enter/press and combined method functionality

---------

Co-authored-by: jsduncan2 <jsduncan2@users.noreply.github.com>

billoley committed:
Fix yielding issues in TLDQueryIterator and AncestorQueryIterator, enable yielding in corresponding integration tests (#3045)

lbschanno committed:
Refactor LimitFields (#1754)

* Refactor LimitFields

Refactor LimitFields and HitsAreAlwaysIncludedTests to address the
following:

- Variable names that are too long and don't make sense.
- Method names that are too long and don't make sense.
- Object usage efficiency.
- Documentation accuracy.

Fixes #1680

* Incoporate changes recommended upon review

* Reapply refactoring

* GitHub Actions: Fix Formatting

* Remove usage of Accumulo Pair

---------

Co-authored-by: Ivan Bella <ivan@bella.name>
Co-authored-by: GitHub Actions <datawave@github.com>
Co-authored-by: Moriarty <22225248+apmoriarty@users.noreply.github.com>

dtspence committed:
IndexInfo serialization support for Kryo (#3037)

* Kryo serialization for IndexInfo objects

* Minor tweaks to the Kryo serialization

* CreateUidsIterator set default serializer to writable for backwards compatibility

* Use member variable for value serializer

* Import order update for formatting

* Update javadoc parameter comment

dtspence committed:
Optimizations related index info evaluation (#3042)

- Set exact size to avoid ArrayList resizing
- Adjust the allowed methods set to be static final

skyler-kv committed:
deprecated previous IncludeExcludeIndexFieldsRule

avgAGB committed:
AOCSV - Adding in conditional for new override format (#3027)

* Adding in a check for new formatting

* Adding check for overrides in new format, some constants

* Tweaking unit tests

alerman committed:
Updating hadoop to 3.4.1 (#2559)

* Updating hadoop to 3.4.0

* Update to hadoop shaded jars

* Update to 3.4.1

* Addressing PR feedback

lbschanno committed:
Feature/dynamic shards (#2691)

* Add support for dynamic shard creation

Add the ability for define custom ShardIdGenerators that can be added
to ShardIdFactory and used to override shard id generation for any
records considered applicable for the generators.

Closes #2632

* GitHub Actions: Fix Formatting

* Add documentation for ShiftOnDay

* Rename constant to be consistent with value

---------

Co-authored-by: GitHub Actions <datawave@github.com>
Co-authored-by: Ivan Bella <347158+ivakegg@users.noreply.github.com>

apmoriarty committed:
Fix Node Lineage in ExpandCompositeTerms (#2883)

* Fix bug in ExpandCompositeTerms that broke node lineage

* Clean up extra reference expressions

avgAGB committed:
Adding geoserver version update (#2988)

hlgp committed:
Add query field rule (#2672)

* Add query field rule

* Add QueryPlanningStage (missed in first commit)

* updates to allow rules to access node lineage

* Make parseRules public

* Delete warehouse/query-core/src/main/java/datawave/query/planner/QueryPlanningStage.java

remove unused class

* Delete warehouse/query-core/src/main/java/datawave/query/jexl/visitors/QueryFieldsRuleVisitor.java

remove unused class

* move members to impl

---------

Co-authored-by: hlgp <hlgp@users.noreply.github.com>
Co-authored-by: Moriarty <22225248+apmoriarty@users.noreply.github.com>

ddanielr committed:
Adds warnings for when regex doesn't match (#3053)

* Adds warnings for when regex doesn't match

Adds warning log messages for when tier regex patterns do not match any
tservers.

This covers the case where tservers match against one of the tier
regexes, but not a second regex.

Removed deprecated useage of deprecated date methods.

* assign no tablets to tier w/ zero tservers

* add comment

---------

Co-authored-by: Keith Turner <kturner@apache.org>

jalphonso committed:
update microservice-parent jkube configuration (#3054)

Co-authored-by: Joe Alphonso <jfalpho@uwe.nsa.gov>

Commit:
[maven-release-plugin] prepare release microservices/microservice-parent-4.0.5

Commit:
[maven-release-plugin] prepare for next development iteration

jalphonso committed:
update parent version in microservice-service-parent (#3056)

* update parent version in microservice-service-parent

* temporarily disable compose snapshot ci job

---------

Co-authored-by: Joe Alphonso <jfalpho@uwe.nsa.gov>

Commit:
[maven-release-plugin] prepare release microservices/microservice-service-parent-5.0.5

Commit:
[maven-release-plugin] prepare for next development iteration

lbschanno committed:
Add query validation configuration for quickstart (#3040)

Currently all attempts to use the Query/{logicName}/validate endpoint to
validate queries in the Quickstart will result in a 'Query validation
not implemented' error due to the lack of configured rules for the
baseline ShardQueryLogic bean.

Add default query validation rule definitions to the quickstart
QueryLogicFactory.xml to fix this.

foster33 committed:
Change sizeInBytes to check for Long MIN_VALUE (#3057)

kmtoth committed:
Update pom's for 7.28.0-SNAPSHOT

jalphonso committed:
cleanup quickstart and jkube docker configuration (#2998)

* cleanup quickstart and jkube docker configuration

set compose prefix

stop boostrapping maven twice

add default back for microservices since they do not load the properties files

add latest tag to compose tests

# Conflicts:
#	contrib/datawave-quickstart/docker/docker-build.sh
#	contrib/datawave-quickstart/docker/pom.xml

# Conflicts:
#	microservices/microservice-parent/pom.xml

* remove cdn download from pom and set file urls to local paths

* fix file paths

* testing

# Conflicts:
#	microservices/microservice-service-parent/pom.xml

* cleanup and add quickstart web tests back to CI

* update quickstart ci test

* testing fixes

* fix bug

* address compose warning

* bugfix

* dev profile clean up and docker-entrypoint.sh fix

* bootstrap only when install is called

* kill process when there is a fatal error

* update maven to a version we actually have

* install maven via os pkg mgr to start

* move quickstart jdk check to the install.sh scripts

* copy maven from latest maven-3 image at container build time

* remove maven download and bootstrap code

* fix returns and exits so we terminate quickstart build upon first fatal error

* remove quickstart ci job

* remove broken maven download from inside quickstart

* remove obsolete arg

* bump microservice-parent version in child poms

* bump microservice-service-parent version in child poms

* use mvn --settings arg

* rename vars and add to pom for warehouse docker image

* add web tests to CI

* reenable compose snapshot ci job

* add missed mvn --settings arg

* testing

* fix bug

* remove testing changes

* fix web tests

* fixes so users can use scripts outside of the container

---------

Co-authored-by: Joe Alphonso <jfalpho@uwe.nsa.gov>

ivakegg committed:
Added patterns to ensure no non-ascii characters are included. (#3069)

Also added pattern for any unescaped 'd'.

apmoriarty committed:
Fix intermittent test failure for ThreadedRangeBundlerIteratorTest (#3059)

foster33 committed:
Prevent Docker Quickstart from using incorrect MAVEN_HOME path (#3065)

The MAVEN_HOME ENV variable set in the Dockerfile is not used in the
COPY / ADD lines. Instead, the host's MAVEN_HOME path is used and will
lead to a "file not found" error if it is anything other than
`/usr/share/maven`.

This resolves this issue by hardcoding the paths instead.

Fixes #3064

jalphonso committed:
Quickstart improvements (#3063)

assertCreateDir bugfix in Quickstart  
align hadoop, zookeeper, accumulo, wildfly versions across the repo

---------

Co-authored-by: Joe Alphonso <jfalpho@uwe.nsa.gov>

jalphonso committed:
update microservice parent to fix docker issue pushing to registry (#3061)

renamed docker.image.prefix to docker.registry to make purpose clear
resolves image push failures

---------

Co-authored-by: Joe Alphonso <jfalpho@uwe.nsa.gov>

Commit:
[maven-release-plugin] prepare release microservices/microservice-parent-4.0.6

Commit:
[maven-release-plugin] prepare for next development iteration

jalphonso committed:
bump microservice-service-parent parent version (#3070)

Co-authored-by: Joe Alphonso <jfalpho@uwe.nsa.gov>

Commit:
[maven-release-plugin] prepare release microservices/microservice-service-parent-5.0.6

Commit:
[maven-release-plugin] prepare for next development iteration

Commit:
[maven-release-plugin] prepare release datawave-code-style_1.0.1

Commit:
[maven-release-plugin] prepare for next development iteration

Commit:
[maven-release-plugin] prepare release assert-properties_2.0.2

Commit:
[maven-release-plugin] prepare for next development iteration

Commit:
[maven-release-plugin] prepare release read-properties_2.0.3

Commit:
[maven-release-plugin] prepare for next development iteration

jalphonso committed:
bump datawave-code-style version (#3074)

Co-authored-by: Joe Alphonso <jfalpho@uwe.nsa.gov>

jalphonso committed:
bump assert and read plugin versions (#3075)

Co-authored-by: Joe Alphonso <jfalpho@uwe.nsa.gov>

jalphonso committed:
fix docker registry handling (#3060)

renamed `docker.image.prefix` to `docker.registry` to make purpose clear
resolves image push failures

---------

Co-authored-by: Joe Alphonso <jfalpho@uwe.nsa.gov>

Commit:
update microservice-parent

Commit:
[maven-release-plugin] prepare release microservices/microservice-parent-4.0.7

Commit:
[maven-release-plugin] prepare for next development iteration

Commit:
update microservice-service-parent

Commit:
[maven-release-plugin] prepare release microservices/microservice-service-parent-5.0.7

Commit:
[maven-release-plugin] prepare for next development iteration

jalphonso committed:
Fix docker push (#3076)

fixing docker push for real this time

---------

Co-authored-by: Joe Alphonso <jfalpho@uwe.nsa.gov>

Commit:
fix warehouse assemble ingest image

apmoriarty committed:
Feature flag bounded range validation post index expansion (#3078)

Feature flag bounded range validation and update test expectations

apmoriarty committed:
DateIndexHelper non-iterator methods should still use a batch scanner (#3079)

When the DateIndexHelper is not using the DateIndexIterator it should
use a batch scanner

alerman committed:
Update sanity check for a wider range of dates. (#2917)

Co-authored-by: GitHub Actions <datawave@github.com>

alerman committed:
Update generate splits (#2760)

Co-authored-by: GitHub Actions <datawave@github.com>

jalphonso committed:
update README (#3080)

if building images locally for docker compose use add
`-Djkube.container-image.tags.1=latest` arg
this is no longer the default as we don't want to publish the latest tag
to the docker registry

Co-authored-by: Joe Alphonso <jfalpho@uwe.nsa.gov>

jalphonso committed:
remove dist arg for quickstart builds (#3081)

quickstart builds do not benefit from building source and javadoc  
remove the dist arg from `DW_DATAWAVE_BUILD_COMMAND` to stop this

Co-authored-by: Joe Alphonso <jfalpho@uwe.nsa.gov>

apmoriarty committed:
LimitFields Updates (#2749)

- LimitFields now considers the backing data when determining if an
attribute is a hit attribute. Previously a PreNormalizedAttribute would
fail comparison against a TypeAttribute despite having the same metadata
key and backing value
- Added LimitFieldsTest and some basic test cases

avgAGB committed:
Task/switch builtin calls (#3025)

Co-authored-by: GitHub Actions <datawave@github.com>

ivakegg committed:
Lossy Numeric Regex (#2589)

original regex for number types

---------

Co-authored-by: hlgp <hlgp@users.noreply.github.com>

foster33 committed:
Create and integrate stat utility for easy formatting (#3072)

Fixes #2909

alerman committed:
Set balancer delay so create shards will go fast in quickstart (#3083)

apmoriarty committed:
Document avoid collection copies (#3082)

There are several Document methods that are copying the underlying
collection of attributes. Stop doing that.

jalphonso committed:
rework compose CI logic (#3084)

CI pipeline will now succeed as long as at least one Compose Job passes
Reusable workflows helped with code de-duplication  
Broke out compose job into meaningful descriptive steps  
Leveraging existing Github action `actions/setup-java` to setup Maven
settings.xml

---------

Co-authored-by: Joe Alphonso <jfalpho@uwe.nsa.gov>

alerman committed:
slf4j and log4j updates (#3085)

Breaking up the Bulkv2 PR to try to limit its scope and identify issues
in smaller batches

ivakegg committed:
Updated Visitor function to avoid modifying config as it needs to remain (#3086)

thread safe

ivakegg committed:
Fixed issue with ivarator building visitor where the directory used was not specific enough (#3090)

to prevent improper reuse.

alerman committed:
Update to maven build cache with some fixes (#3096)

Always run spring repackage
Always run docker builds if necessary
Consider changes to dockerfiles/yaml/json files
reenable
exclude target and logs

apmoriarty committed:
Range and regex filter iterators should only seek the underlying sour… (#3095)

…ce when a move() is called

ivakegg committed:
Updated to better invalidate illegal characters (#3097)

alerman committed:
Bulk v2 accumulo 4894 rebased (#3092)

Co-authored-by: keith-ratcliffe <ratcliffe.keith@gmail.com>
Co-authored-by: Daniel Roberts ddanielr <ddanielr@gmail.com>

hgklohr committed:
Update pom's for 7.29.0-SNAPSHOT

dws4 committed:
Filter Button to Retrieve Prior 'X' Amount of Days (#3087)

This PR does the following:
1. Creates a filter icon near the search bar
2. Can filter 'X' days prior based on a full search _(by the search
bar)_ by input and 'apply' button.
3. Can reset and show all rows by clicking the 'show all rows' button by
showing all rows.
4. On loading, it will default to only 30 days prior.
5. _NEW 8/13/25:_ Fixed the Button Parse Bug (duplicate buttons
appearing on same sections of data).

_Note: ~~Right now this has not been integration tested~~. ~~Probably
needs a rebase as well.~~_

---------

Co-authored-by: foster33 <84727868+foster33@users.noreply.github.com>

ivakegg committed:
Updated to avoid moving to far when yielding (#3104)

in the OrIterator.moveIterators call

---------

Co-authored-by: Bill Oley <bill.oley@gmail.com>

alerman committed:
Add remote maven cache (#3101)

Co-authored-by: Adam Lerman <adam.lerman@red-drum.com>

foster33 committed:
Make query metric starter version name consistent (#3105)

The naming of the query metric starter version was inconsistent with
what is used elsewhere ([see query-starter
pom](https://github.com/NationalSecurityAgency/datawave/blob/674fb299b03dbf5d162dd2b0ec241071e00fa2f5/microservices/starters/query/pom.xml#L31)).
It also was inconsistent with the naming format used by other starters.

apmoriarty committed:
Exercise yielding in NestedIterators (#3109)

Refactor TestWaitWindowObserver to accept a percentage chance of
yielding instead of a modulus denominator

jack-gitdev committed:
Add usage of configurable property to remove types when processing un… (#3107)

…fielded terms. This PR is the integration based version related to PR
#3103

hgklohr committed:
Revert "Update generate splits (#2760)"

This reverts commit 5b8f9af2cb8d350f72b5846c74710cd2053d7197.

hgklohr committed:
Revert "Set balancer delay so create shards will go fast in quickstart (#3083)"

This reverts commit 1a754298ec66ea7cd52ab6441039de54f738675f.

avgAGB committed:
Dependabot dnsjava-2 version updates (#3030)

Co-authored-by: Connor <connor@connortech.me>

dws4 committed:
Data Dictionary Help Menu (#3115)

This creates a Data Dictionary Help Menu that allows users to click
between 3 menu options. It does the following:
1. Question Help Menu in the Top right _(circular button)_. Follows the
User's page when scrolling to account for the banner.
2. Opens to 3 menu options.
3. Opens a new tab for the user to these 3 links.

skyler-kv committed:
validation - new rule to interpret grouped fielded phases (#3088)

`GroupedInterpretationRule` will give a message when a grouped query is
something similar to `FOO:(aaa bbb ccc)`, `(FOO:aaa bbb cc)`, or
`(FOO:aaa AND FOO:bbb AND FOO:ccc)`

i.e. if the user queries `FOO:(aaa bbb ccc)`, the following message
would occur
"Operator precedence may be missing, field(s) `[FOO]` with value(s)
`[aaa, bbb, ccc]` will be interpreted as `( FOO:aaa AND FOO:bbb AND
FOO:ccc )`"

Fixes #2874

billoley committed:
Query metric service should use DefaultQueryPlanner instead of DatePartitionedQueryPlanner

Commit:
[maven-release-plugin] prepare release microservices/services/query-metric/service-4.1.6

Commit:
[maven-release-plugin] prepare for next development iteration

skyler-kv committed:
Merge branch 'integration' into validationindex
